pkg/controllers/disruption/types.go

```diff
@@ -71,6 +72,10 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events
 	if !node.Initialized() {
 		return nil, fmt.Errorf("state node isn't initialized")
 	}
+	// If the orchestration queue is already considering a candidate we want to deprovision, don't consider it a candidate.
+	if err := queue.CanAdd(node.Name()); err != nil {
```

Comment:
```suggestion
	if err := queue.CanAdd(node.ProviderID()); err != nil {
```

Do we have a test for this? This is wrong, this needs to be the provider id

Reply:
Good catch